Add rt1180 i3c support

lucien-nxp opened the pull request:
Support p3t1755ddp_ard_i3c/p3t1755ddp_ard_i2c shield boards on thermometer sample case using i3c2/lpi2c2 instance.

lucien-nxp committed:
dts: arm: nxp: add two i3c instances for RT1180

add i3c instances
enable i3c clock under soc folder

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
drivers: clock_control: add i3c clock for clock_control_mcux_ccm_rev2.c

add i3c case to get i3c instance clock

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
boards: nxp: mimxrt1180_evk: add i3c2/lpi2c support for p3t1755dp_ard

add the i3c2 pin setting
set i3c2 instance as ok status
add p3t1755dp_ard_i3c_interface/p3t1755dp_ard_i2c_interface node label

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

danieldegrasse commented on drivers/clock_control/clock_control_mcux_ccm_rev2.c:
```diff
@@ -74,6 +74,13 @@ static int mcux_ccm_get_subsys_rate(const struct device *dev,
 #endif
 #endif
 
+#ifdef CONFIG_DT_HAS_NXP_MCUX_I3C_ENABLED
```
Why use the DT Kconfig versus CONFIG_I3C_MCUX?

lucien-nxp replied:
Sorry, I didn't think too much about it. CONFIG I3C MCUX looks more appropriate.

danieldegrasse commented on include/zephyr/dt-bindings/clock/imx_ccm_rev2.h:
```diff
@@ -139,6 +139,11 @@
 #define IMX_CCM_MIPI_CSI2RX_UI_CLK   0x2000UL
 #define IMX_CCM_MIPI_CSI2RX_ESC_CLK  0x2100UL
 
+/* I3C */
```
Don't overload IDs like this- use a new set of MSBs, maybe `0x2200UL`?

danieldegrasse commented on samples/sensor/thermometer/boards/mimxrt1180_evk_mimxrt1189_cm33.overlay:
```diff
@@ -0,0 +1,34 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* The p3t1755 shield board is plugged into the arduino interface on
```
Style nit- make these comments start on newline, like so:
```
/*
 * comment
 */
```

lucien-nxp replied:
Updated.

danieldegrasse commented on samples/sensor/thermometer/boards/mimxrt1180_evk_mimxrt1189_cm33.overlay:
```diff
@@ -0,0 +1,34 @@
+	i3c-scl-hz = <DT_FREQ_M(4)>;
+	i3c-od-scl-hz = <DT_FREQ_K(1500)>;
+
+	p3t1755: p3t1755@4800000236152a0090 {
```
Just a question- should we be defining this as a shield? It seems like we might reuse this sensor for I3C enablement on a bunch of platforms, right?

lucien-nxp replied:
1. The name "shield" wasn't defined by me. NXP release the P3T1755DP Arduino® Shield Evaluation Board officially on [NXP web](https://www.nxp.com/design/design-center/development-boards-and-designs/analog-toolbox/arduino-shields-solutions/p3t1755dp-arduino-shield-evaluation-board:P3T1755DP-ARD)
2. Yes, we can use this board on a bunch of platforms, if exist arduino interface and standalone I3C interface on nxp board.

danieldegrasse replied:
Makes sense. We don't need to do this as part of this PR, but we should consider defining this as a shield to simplify future enablement- especially because it is using the Arduino interface. Is the I3C bus on the RT1180 exposed on the same pins Arduino typically uses for I2C?

lucien-nxp replied:
Thank you for pointing out. I have added p3t1755dp_ard shield board and support shield board on RT1180 board.

lucien-nxp committed:
samples: sensor: thermometer： add shield description

add mimxrt1180-evk board support description
add shield parameter description for RT1180 board by i3c/i2c ways

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

hakehuang commented on samples/sensor/thermometer/sample.yaml:
```diff
@@ -18,3 +18,18 @@ tests:
+    filter: dt_alias_exists("ambient-temp0")
+    extra_args: SHIELD=p3t1755dp_ard
+    platform_allow:
+      - mimxrt1180_evk/mimxrt1189/cm33   # p3t1755
```
do not add this

lucien-nxp replied:
Thank you for pointing out the issue. I have updated in yml.

hakehuang commented:
test PASS
```
Temperature is 13.0°C
Temperature is 13.0°C
Temperature is 13.0°C
Temperature is 13.0°C
Temperature is 13.0°C
Temperature is 13.0°C
Temperature is 13.0°C
*** Booting Zephyr OS build v4.0.0-1264-g516dc33f5e73 ***
Thermometer Example (arm)
Temperature device is 0x38013a04, name is p3t1755@4800000236152a0090
Temperat*** Booting Zephyr OS build v4.0.0-1269-g899b5465d6f9 ***
Thermometer Example (arm)
Temperature device is 0x38013a04, name is p3t1755@4800000236152a0090
Temperature is 13.0°C

```

danieldegrasse reviewed (dismissed):
LGTM- from what I can tell, this shield supports I3C or I2C, but we are using the I3C connection (which is a custom header, and does not match the arduino I2C pinout). That's all correct, right?

lucien-nxp commented:
> LGTM- from what I can tell, this shield supports I3C or I2C, but we are using the I3C connection (which is a custom header, and does not match the arduino I2C pinout). That's all correct, right?

Yes, Arduino interface only supply the power, and the communication with sensor is used  the I3C/I2C connection.

hakehuang commented:
@Lucien-Zhao , I am sorry for the missing leading but I find something wrong here, even I do not connect the shield board the thermo sample can still pass, and in some check I find the below issue, 

```
according to the p3t shield board design the i2c address is 0x4C not ox48
```
see below

![image](https://github.com/user-attachments/assets/75bf5713-45a4-481e-a3d9-70acdcee5b3f)

and the board default setting

![image](https://github.com/user-attachments/assets/8e3e69d3-4cd2-4b59-99b0-3a2de8c39e71)

and the interesting thing is that the 0x48 address do has response from the board.

hakehuang commented:
> Temperature is 13.0°C

it always read back 13.0°C, and if you enable the I3C log, there are many errors

kartben commented:
Needs rebase

lucien-nxp committed:
dts: arm: nxp: add prescaler parameter in dts

prescaler parameter had been set to required true, add
this parameter to resolve the building issues.

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
boards: shields: Add shield for P3T1755DP-ARD-I2C modules

Add a shield for P3T1755DP-ARD-I2C modules.
Support the shield on MIMXRT1180-EVK board by I2C way.

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
boards: shields: Add shield for P3T1755DP-ARD-I3C modules

Add a shield for P3T1755DP-ARD-I3C modules.
Support the shield on MIMXRT1180-EVK board by I3C way.

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

lucien-nxp committed:
drivers: sensor: nxp: add code judge whether read back correctly

find a sensor driver bug don't judge the data whether read back
correctly, deal with buffer data directly.

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

hakehuang reviewed (dismissed):
board testing pass for I3C and I2c, and the baudrate can only set to 400K for shield board cable, above this, the read will not be stable.

dleach02 commented:
I did a rebase to force a clean CI

lucien-nxp commented:
Update two shields index.rst to resolve the doc issues.

dleach02 commented:
@hakehuang if your testing is good can you +1
